hostpool.go

```diff
@@ -214,18 +275,26 @@ func (p *standardHostPool) markFailed(hostR HostPoolResponse) {
 	host := hostR.Host()
 	p.Lock()
 	defer p.Unlock()
+
 	h, ok := p.hosts[host]
 	if !ok {
 		log.Fatalf("host %s not in HostPool %v", host, p.Hosts())
 	}
 	if !h.dead {
-		h.dead = true
-		h.retryCount = 0
-		h.retryDelay = p.initialRetryDelay
-		h.nextRetry = time.Now().Add(h.retryDelay)
+		if h.failures != nil {
+			ts := time.Now()
+			h.failures.insert(ts)
+			if h.failures.since(ts.Add(-p.failureWindow)) > p.maxFailures {
+				log.Printf("host %s exceeded %d failures in %s", h.host, p.maxFailures, p.failureWindow)
```

Comment:
stray log

Reply:
As discussed, let's leave this in (although it'd be nice if this package allowed you to configure a logger).